lib/fast_jsonapi/object_serializer.rb

```diff
@@ -314,7 +309,13 @@ def validate_includes!(includes)
             relationships_to_serialize = klass.relationships_to_serialize || {}
             relationship_to_include = relationships_to_serialize[parsed_include]
             raise ArgumentError, "#{parsed_include} is not specified as a relationship on #{klass.name}" unless relationship_to_include
-            klass = relationship_to_include.serializer.to_s.constantize unless relationship_to_include.polymorphic.is_a?(Hash)
+            if relationship_to_include.static_serializer
+              klass = relationship_to_include.static_serializer
+            else
+              # the serializer may change based on the object (e.g. polymorphic relationships),
+              # so inner relationships cannot be validated
+              break
```

Comment:
Let's add a test for this use-case. We might need to actually just skip (`next`) instead of cancelling the validation of the remaining records (or I'm missing the context here, please correct/help me understand).

Reply:
This was also reported and fixed in #13.  Personally I would prefer to remove this method all together.  All it does is pre-validate the `include` option prior to serialization.  To me, this is not helpful at all because the serialization will fail anyway.  It is extra upfront work (and therefore a performance hit) every time a serialization occurs, and it cannot even be validated if the serializer is dynamic.  I think we are better off just assuming the `include` option is valid and let the serializer fail normally if not.

Before this change, it would incorrectly throw an exception for dynamic serializers (e.g. `polymorphic`).  I minimized the change while avoiding breakage by simply `break`ing instead of failing.

If we were to write a test case for this scenario, the correct test case would simply prove that this validation does not work correctly with dynamic serializers.  I would rather solve the problem a different way.

Maybe we can move the discussion over to #13 or start a different issue.  In the meantime, I would like to just have this bug fixed.

Reply:
Alright, let's do it in the next PR.